### ISSUE TITLE
Autosc gets stuck and won't respond until a manual WS is performed

### ISSUE THREAD

**Issue by ZaphodJustSomeGuy** [ekrividus/autoSC#4]
I have been using this addon for a few weeks now across 2 accounts on separate computers.  Typically I setup 1 account to open with a particular WS and then both can respond to close step 2 or more of a chain.

There is some particular condition that causes at least one of the characters to stop responding and then just sits there with 3000TP, even on a fresh mob.  Only after a manual intervention of using up the TP by doing any WS does autosc spring back into life.  This usually happens after a quite some time but doesn't seem to be at a fixed time after starting //asc on. 

So far I have not been able to figure out what the condition is that causes this.  I am usually using this in conjunction with a few other addons like autocor and easyfarm and it could be that more than 1 addon tries to do something at the same time and causes autosc to kind of lockup.

I'm not sure if this addon is still supported or not.  It's a great shame to setup an overnight EP session that's getting 50-60k EP/hour only to find it dropped to 25k/hour because 1 char is no longer doing a WS.

**Comment by ZaphodJustSomeGuy** [comment]
I would be willing to help try to track down the issue if there is ongoing development of this addon.  Perhaps by looking back through my logs I can see some recurring action that happens before the lockup.

**Comment by ekrividus** [comment]
I haven't played in a couple of years so testing and debugging hasn't been kept up. Though this addon has been very stable in its current state so that hasn't been a real issue. I'm happy to take a look at things though and see if we can sort it out together.

If you can turn on some of the debug options. I know this bit will be kind of a lot to ask but to really nail down what's happening we need to debug info from when that first WS fails to fire which probably means watching things. I suppose you could record an overnight session and just watch what happens. I suspect that the stuck instance will probably show debug info that it is currently waiting for an existing SC window that has actually expired but it never saw the packet.

My guess here though is that it isn't actually the addon so much but that anytime you have lots of things reading/consuming packets across multiple accounts on a single machine sometimes strangness happens and a packet will just kind of get "stuck" or lost and it'll bork things. It's like when CurePlease stops curing or Scripted stops targeting mobs. I believe this is largely one of those things that comes from running a game that wasn't ever intended to run in multiple instances in many instances, the instances aren't as sandboxed as they really need to be which isn't really the fault of anyone, just the age of the game.

All that said. I can offer a couple of workarounds that wouldn't need much/any testing or waiting on me to patch :D cause I'm slow. 
Setup the instance of autoSC that is freezing to open SCs, but only when it's over like 2800 TP. That way if it hits 3k it'll try to open and make sure that it ignores ongoing SC effects for openers. 
You could also setup EF to do the same, for any/all instances have it fire a random WS if you have 3k TP that way nobody gets stuck on 3k.

**Comment by ZaphodJustSomeGuy** [comment]
Thanks for your quick reply.  Yeah I was thinking the same about the workarounds, it is rare enough that it's hard to spot but is sporadic so happen 1-2 hours in or after 12 hours.  I think your explanation around lost packets is probably the key, a version of that can happen in the logs and screw up things like healbot etc for a few seconds until it catches the next valid log entry.

I'll do a workaround like you said.

Btw I have another issue around the filter command not working, but I'll put that in a different issue heading in a minute.

**Comment by ekrividus** [comment]
And the more I thought on this the more I think I remember it happening before. I'd run 6 chars all w/ various things running and occasionally notice that WSs would stop firing. I think I ended up disabling the existing SC check but had everyone open at 3k except the regular opener that would still go at 1k. Not a perfect solution but a viable work around.
I feel like if I come back to the game and Windower 4 is still the way I may just fix this. Or who knows do a full rewrite lmao.
